packages/riipen-ui/src/components/Table.jsx

```diff
@@ -0,0 +1,241 @@
+                ))}
+              </TableRow>
+            );
+          })}
+        </TableBody>
+      </React.Fragment>
+    );
+  }
+
+  renderMobile(tableContent, entities) {
+    const { hover } = this.props;
+
+    const primary = Object.entries(tableContent).find(
+      ([, values]) => values.primary
+    );
+
+    let bodyContent = {};
+    let primaryContent;
+    let primaryKey;
+
+    if (primary) {
+      [primaryKey, primaryContent] = primary;
+      bodyContent = Object.keys(tableContent).reduce((object, key) => {
+        if (key !== primaryKey) {
+          object[key] = tableContent[key];
+        }
+        return object;
+      }, {});
+    }
+
+    const entityCount = entities.length;
+
+    return (
+      <React.Fragment>
+        <TableBody>
+          {entities.map((entity, i) => {
+            return (
+              <TableRow
+                hover={hover}
+                border={i !== entityCount - 1}
+                key={`entity-${i}`}
+              >
+                <TableCell spacing={0}>
+                  {primaryContent && (
+                    <TableContainer background={false}>
+                      <TableBody>
+                        <TableRow>
+                          <TableCell span={2}>
+                            {primaryContent?.cell(entity)}
+                          </TableCell>
+                        </TableRow>
+                      </TableBody>
+                    </TableContainer>
+                  )}
+                  <TableContainer centered fixed background={false}>
+                    <TableBody>
+                      {Object.keys(bodyContent).map((key, j) => (
+                        <TableRow border={false} key={`${i}-${j}`}>
+                          <TableHeaderCell>
+                            {tableContent[key]?.label ||
+                              tableContent[key]?.header(entity)}
+                          </TableHeaderCell>
+                          <TableCell>
+                            {tableContent[key]?.cell(entity)}
+                          </TableCell>
+                        </TableRow>
+                      ))}
+                    </TableBody>
+                  </TableContainer>
+                </TableCell>
+              </TableRow>
+            );
+          })}
+        </TableBody>
+      </React.Fragment>
+    );
+  }
+
+  render() {
+    const { isMobile } = this.state;
+    const { entities, children, classes } = this.props;
+
+    const theme = this.context;
+
+    let nodes = children;
+    if (!children.length) {
+      nodes = [children];
+    }
+
+    const tableContent = nodes.reduce(
+      (acc, child) =>
+        Object.assign(acc, {
+          [child.key]: {
+            label: child.props.label,
+            primary: child.props.primary,
+            header: child.props.header,
+            cell: child.props.cell
+          }
+        }),
+      {}
+    );
```

Comment:
so the reason I used key was because it was something that would be required in this case as you are creating a list of nodes in the parent component. I am fine with creating our own, but it would kind of be like adding a second key prop that would be used just for us and would 99% of the time be the exact same as the key value. 

This is the area I expected the most push back. I actually really like the way this is implemented as from the outside it is very easy to read. We could move all of this into a prop of the table component and just have one giant prop list but to me this feels better.